test/models/sync_plan_test.rb

```diff
@@ -129,5 +129,25 @@ def test_schedule_format_disabled
       assert_match(/\/PT24H$/, schedule)
       assert_includes schedule, @plan.sync_date.iso8601
     end
+
+    def test_add_product
+      product = katello_products(:redhat)
+      @plan.products << product
+      @plan.valid?
+    end
+
+    def test_remove_product
```

Comment:
What about just testing empty plans? The `@plan` object starts with no products, right? Why not just test:

```rb
assert(@plan.valid?, "Empty plan must be valid")
```

with no adding/clearing products?